stan/math/prim/scal/fun/get_num_threads.hpp

```diff
@@ -0,0 +1,76 @@
+#ifndef STAN_MATH_PRIM_SCAL_FUN_GET_NUM_THREADS_HPP
+#define STAN_MATH_PRIM_SCAL_FUN_GET_NUM_THREADS_HPP
+
+#include <stan/math/prim/scal/err/invalid_argument.hpp>
+#include <boost/lexical_cast.hpp>
+
+#include <cstdlib>
+#include <thread>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Get number of threads to use for num_jobs jobs. The function uses
+ * the environment variable STAN_NUM_THREADS and follows these
+ * conventions:
+ *
+ * - STAN_NUM_THREADS is not defined => num_threads=1
+ * - STAN_NUM_THREADS is positive => num_threads is set to the
+ *   specified number
+ * - STAN_NUM_THREADS is set to -1 => num_threads is the number of
+ *   available cores on the machine
+ * - STAN_NUM_THREADS < -1, STAN_NUM_THREADS = 0 or STAN_NUM_THREADS is
+ *   not numeric => throws an exception
+ *
+ * Should num_threads exceed the number of jobs, then num_threads will
+ * be set equal to the number of jobs.
+ *
+ * @param num_jobs number of jobs
+ * @return number of threads to use
+ * @throws std::runtime_error if the value of STAN_NUM_THREADS env. variable
```

Comment:
This is more of a final review than what you need right now, but this throws an std::invalid_argument.

Reply:
oh, you are right... will fix.